modules/renter.go

```diff
@@ -83,6 +83,38 @@ type FileInfo struct {
 	Expiration     types.BlockHeight `json:"expiration"`
 }
 
+// FileDetailInfo provides detail information about a file.
+// first dimension of details is chunk id, second is chunk redundency hosts
+type FileDetailInfo struct {
+	SiaPath        string            `json:"siapath"`
+	Filesize       uint64            `json:"filesize"`
+	Available      bool              `json:"available"`
+	Renewing       bool              `json:"renewing"`
+	Redundancy     float64           `json:"redundancy"`
+	UploadProgress float64           `json:"uploadprogress"`
+	Expiration     types.BlockHeight `json:"expiration"`
+	Details        *FileDetail       `json:"details"`
```

Comment:
This doesn't need to be a pointer. Also, `Details` isn't a very descriptive name.

I'm wondering if perhaps we should get rid of the duplication between `FileInfo` and `FileDetailInfo`. It's not too bad if you have to call two separate API endpoints. I think I would prefer:
```go
type FileDetailsInfo struct {
    Hosts []HostDetail `json:"hosts"`
}
```